lib/api.js

```diff
@@ -1,48 +1,77 @@
 const { config } = require('./config');
 const request = require('request-promise').defaults({
     method: 'POST',
-    baseUrl: 'https://apptoogoodtogo.com/index.php/api_tgtg',
-    json: true,
+    baseUrl: 'https://apptoogoodtogo.com/api',
     headers: config.get('api.headers')
 });
 const session = {};
 
 module.exports = {
     login,
+    refreshToken,
     listFavoriteBusinesses
 };
 
 function login(credentials = config.get('api.credentials')){
     return request({
-        url: '/login',
-        form: { 
-            email: credentials.email, 
-            password: credentials.password 
+        url: '/auth/v1/loginByEmail',
+        json: {
+            device_type: "UNKNOWN",
+            email: credentials.email,
+            password: credentials.password
+        },
+        headers: {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Accept-Language": "en-US"
         }
-    })
-        .then(handleError)
-        .then(createSession);
+    }).then(createSession);
 }
 
-function handleError(response){
-    if(response.msg === 'OK'){
-        return response;
-    } else {
-        throw new Error(response.msg);
+function refreshToken(refresh_token = session.refresh_token){
+    if (refresh_token == undefined) {
```

Comment:
I would do this the other way around: always call `login` and have login do the check for the refresh token, and return refreshToken when there is one.

Reply:
You're absolutely right. I'll update the pull request.